compare.go

```diff
@@ -6,6 +6,54 @@ import (
 	"github.com/segmentio/parquet-go/deprecated"
 )
 
+// CompareDescending constructs a comparison function which inverses the order
+// of values.
+//
+//go:noinline
+func CompareDescending(cmp func(Value, Value) int) func(Value, Value) int {
+	return func(a, b Value) int { return -cmp(a, b) }
+}
+
+// CompareNullsFirst constructs a comparison function which assumes that null
+// values are smaller than all other values.
+//
+//go:noinline
+func CompareNullsFirst(cmp func(Value, Value) int) func(Value, Value) int {
+	return func(a, b Value) int {
+		switch {
+		case a.IsNull():
+			if b.IsNull() {
+				return 0
+			}
+			return -1
+		case b.IsNull():
+			return +1
+		default:
+			return cmp(a, b)
+		}
+	}
+}
+
+// CompareNullsLast constructs a comparison function which assumes that null
+// values are greater than all other values.
+//
+//go:noinline
+func CompareNullsLast(cmp func(Value, Value) int) func(Value, Value) int {
+	return func(a, b Value) int {
+		switch {
+		case a.IsNull():
+			if b.IsNull() {
+				return 0
+			}
+			return +1
+		case b.IsNull():
+			return -1
+		default:
+			return cmp(a, b)
+		}
+	}
+}
+
```

Comment:
Moved from `search.go` as it seemed more intuitive to find these functions in `compare.go`.